llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -806,13 +810,15 @@ class AMDGPUImageDimIntrinsic<AMDGPUDimProfile P_,
       !if(P_.IsSample, [llvm_v4i32_ty,           // samp(SGPR)
                         llvm_i1_ty], []),        // unorm(imm)
       [llvm_i32_ty,                              // texfailctrl(imm; bit 0 = tfe, bit 1 = lwe)
-       llvm_i32_ty]),                            // cachepolicy(imm; bit 0 = glc, bit 1 = slc, bit 2 = dlc)
+       llvm_i32_ty],                             // cachepolicy(imm; bit 0 = glc, bit 1 = slc, bit 2 = dlc)
+      !if(!or(P_.IsAtomic, P_.IsAtomicLoad), [llvm_i32_ty], [])),       // atomic ordering
```

Comment:
We should try to avoid breakage caused by changing the function signature of these intrinsics. How about adding the ordering to the cachepolicy field?

Reply:
Yes, good idea. Thanks for the suggestion.

Reply:
Switched to using cachepolicy field instead of ordering parameter.